### ISSUE TITLE
only images or also views?

### ISSUE THREAD

**Issue by xpepermint** [harism/android-pagecurl#13]
Hey, how can I use this lib with LinearLayout instead of images (BitmapProvider)?
Any code examples. Thx!

**Comment by CarlLee** [comment]
I've seen the code, and it's theoretically easy to use Views for CurlView, too.
You can use a Canvas and a Bitmap to draw any Views you would like to be curled on.
Something like this:
 Canvas canvas = new Canvas();
 Bitmap bitmap = Bitmap.createBitmap(width, height, new Config());
 canvas.setBitmap(bitmap);
 myView.draw(canvase);

then, you got a Bitmap and you know how to use it!

**Comment by thiagolocatelli** [comment]
Did you manage to get Views working with CurlView?

**Comment by novatoide** [comment]
I don´t understand very well CarlLee´s comentary. Somebody knows how to do the page curl between layouts? Thank you

**Comment by thiagolocatelli** [comment]
Exactly, i also do not understand CarlLee's comment, it would be good if he could give a better explanation on it.

**Comment by CarlLee** [comment]
I haven't tried this myself, but I am just presenting a way to draw a View onto a Canvas object where the Canvas is bound to a new Bitmap object to store whatever is drawn onto the Canvas.
Every View object has a .draw() method which I believe is what the UI framework calls when it needs to draw another frame of the View object.

```
Canvas canvas = new Canvas();
Bitmap bitmap = Bitmap.createBitmap(width, height, new Config());
canvas.setBitmap(bitmap);
```

those three lines prepared the Canvas and the Bitmap to store whatever is drawn onto Canvas.

```
myView.draw(canvas);
```

this line draws the current state of the View(myView, is an instance of View class or its sub-classes)
After drawn the view onto canvas, the Bitmap we prepare previously has a graphic representation( but only one frame) of "myView", which can be used by page-curl library.

**Comment by jaykumarhk1** [comment]
How can i add videoview or play video on page?
